metaflow/plugins/argo/test/test_trigger_live_run.py

```diff
@@ -0,0 +1,265 @@
+from metaflow.client.trigger_live_run import trigger_live_run
+from metaflow.exception import MetaflowException
+import time
+from unittest import TestCase
+
+
+"""
+Tests Outline:
+    - Parameter Passing (flow name "LiveRunParamFlow")
+        - successful parameter passing
+        - wrong type parameter passing (eg, type=set)
+    - Flow Identification (flow name "LiveRunSimpleFlow")
+        - id by flow name
+        - id by template name
+        - id by both (matching)
+        - id by both (mismatch)
+        - id by neither flow name nor template name 
+    - Plugin Name (all other tests correctly use 'Argo')
+        - None
+        - 'FakePlugin'
+    - Other Tests
+        - timeout error (flow takes too long - set wait_timeout to 0)
+        - failure flow (flow name "LiveRunFailureFlow")
+        - missing flow (flow name "LiveRunMissingFlow")
+        - flow that triggers a flow (flow name "LiveRunFlowTriggeringFlow")
+    - Features/aspects tested but w/o their own named tests
+        - Wait Feature - no separate tests because both states (True & False)
+            are incorporated into Flow Identification section
+"""
```

Comment:
Does all integration tests here rely on the corresponding argo workflow template to exist in the cluster first? 

Ideally the test should:
- Not rely on cluster state for testing. Which requires each test run to 
  1. setup test resources (upload argo workflow template)
  2. run tests
  3. clean up temporary test resources (delete workflows) every time the test is ran
- Allow multiple test runs at the same time (pending on workflow template versioning), or alternatively you can identify each workflow template using some unique id like test trigger time /  test pipeline id etc.


Feel free to create backlog tickets in Jira for them if you feel like these will take significant time / you can't get to them by end of the internship.

Reply:
I made a Jira task for this: [AIP-6585 - Make tests not reliant on cluster state and allow concurrent tests](https://zbrt.atl.zillow.net/browse/AIP-6585)